### PR TITLE
Add improvements for List and Find Commands 

### LINKED ISSUE #186: [PE-D][Tester B] Error message shown for excess invalid parameter is inappropriate
In the situation as shown below, the error message says that not supported tag is detected when there is no such excess tags from the input. 

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: OVReader/ped#5

### LINKED ISSUE #211: As a user, I want to gather a summary of my expenditure over the last N days
so that I can narrow down my scope of listing my transactions to specific situations, e.g. my transactions over the past 1 week

### LINKED ISSUE #212: As a user, I want to display my transactions and statistics in chronological order of date
so that I can find my most recent transactions quickly

### LINKED ISSUE #221: Code review for @chydarren
# @chydarren

These `X-PERCENT` constants are a bit strange. While wanting to avoid magic numbers is understandable, maybe give them some more semantically informative names, e.g. `VERY_HIGH_SAVINGS_RATE`, `HIGH_SAVINGS_RATE`, etc. `static final int THREE = 3` is still a magic number :-)

https://github.com/AY2223S1-CS2113-W12-2/tp/blob/b53a3c2bc91adbdeaaf886db361511e8925b02f1/src/main/java/seedu/duke/data/TransactionList.java#L337-L352

Besides this, the code is overall well-commented and easy to read.

### PR DESCRIPTION
**Pull request by chydarren**
This pull request seeks to add numerous improvements to the **List and Find Commands**, as well as to address some of the suggestions/issues produced during the PE Dry Run. Minor enhancements have been implemented to ensure certain safeguards while addressing the bugs.

Changes summarized for this PR as follow:

1. Add a new exception to handle cases whereby List and Stats command tags are incorrectly combined together in input, identified in https://github.com/AY2223S1-CS2113-W12-2/tp/issues/186.
2. Feedback acquired from https://github.com/AY2223S1-CS2113-W12-2/tp/issues/221 has been worked on.
3. Minor refactoring of functions for Find Command class and introduce logging messages.
4. Enable sorting of transactions and statistics via a sorting comparator for `Transaction` objects by their dates in ascending order (fulfills user story - https://github.com/AY2223S1-CS2113-W12-2/tp/issues/212).
5. Implement ability to gather a summary of expenditure over the last N days (fulfills user story - https://github.com/AY2223S1-CS2113-W12-2/tp/issues/211) which was opened due to a feedback from PED that users are unable to get past transactions in the current week when using the last N weeks tags.
6. Implement spending habit comment in `stats s/time_insights` for consistency with `stats s/monthly_expenditure`.
7. Add and update test cases for functions that belong to me in Transaction and TransactionsList classes.
8. The `GlobalNonNumericIndexException` class was renamed to `GlobalIndexNotNumericException` class for conformity to names used for other exception classes.

Due to above implementation, more edge cases have been added into the UI test file. I will update the user guide and developer guide to reflect the changes made.

### CODE REVIEW

File: src/test/java/seedu/duke/data/TransactionListTest.java [hunk 1]
```diff
@@ -78,4 +86,121 @@ public void calculateMonthlyTotalExpense_multipleDateDifferentYear_expectedCorre
...
+    public void findTransactions_noMatchingTransaction_expectEmptyTransactionList() {
+        TransactionList transactions = new TransactionList();
+        transactions.addExpense("buy_an_apple", 1, "fruits",
+                LocalDate.of(2001, 1, 1));
+        transactions.addIncome("sell_a_pear", 2, "fruits",
+                LocalDate.of(2002, 1, 1));
+
+        String transactionsList = transactions.findTransactions("");
+        assertEquals(transactionsList, "");
+    }
```
Comment (wcwy) [inline src/test/java/seedu/duke/data/TransactionListTest.java:113]: This test case over here seems to me that it's testing on an empty string instead of "no matching transaction" as claimed by the method name? Perhaps there can be another separate test case here with input that has really no match, such as "orange" here.

File: src/main/java/seedu/duke/Duke.java [hunk 2]
```diff
@@ -42,6 +42,6 @@ public void run() {
     }
 
     public static void main(String[] args) {
-        new Duke().run(); // TODO: Add a file path when implementing storage feature
```
Comment (wcwy) [inline src/main/java/seedu/duke/Duke.java:45]: Thanks for removing this! It's about time for us to ensure that all TODO comments or any commented code are removed!

**Review by wcwy — APPROVED** [review]
LGTM!
